dance/dance_pipeline.py

```diff
@@ -106,3 +113,55 @@ def _generate_molecules_from_database(self) -> oechem.OEMol:
                 mol = oechem.OEMol()
                 oechem.OEReadMolecule(ifs, mol)
                 yield mol
+
+    def assign_fingerprint(self,
```

Comment:
@jmaat This method is the main part of the PR.

Reply:
Visit https://dance.readthedocs.io/en/btjanaka-fingerprint/api/dance.DancePipeline.html#dance.DancePipeline.assign_fingerprint for the rendered version of the docstring.